go/oasis-node/cmd/registry/entity/entity.go

```diff
@@ -314,61 +387,69 @@ func doRegisterOrDeregister(cmd *cobra.Command, args []string) {
 			time.Sleep(1 * time.Second)
 		}
 	}
-
 	os.Exit(1)
 }
 
-func doRegister(client grpcRegistry.EntityRegistryClient, ent *entity.Entity, signer signature.Signer) error {
+func doSignRegister(txFilePath string, ent *entity.Entity, signer signature.Signer) (*serializedTx, error) {
 	ent.RegistrationTime = uint64(time.Now().Unix())
 	signed, err := entity.SignEntity(signer, registry.RegisterEntitySignatureContext, ent)
 	if err != nil {
 		logger.Error("failed to sign entity",
 			"err", err,
 		)
-		return err
+		return nil, err
+	}
+
+	tx := &serializedTx{
+		Register: signed,
+	}
+	return tx, nil
+}
+
+func doSignDeregister(txFilePath string, ent *entity.Entity, signer signature.Signer) (*serializedTx, error) {
+	ts := registry.Timestamp(time.Now().Unix())
+	signed, err := signature.SignSigned(signer, registry.DeregisterEntitySignatureContext, &ts)
+	if err != nil {
+		logger.Error("failed to sign deregistration",
+			"err", err,
+		)
+		return nil, err
 	}
 
+	tx := &serializedTx{
+		Deregister: signed,
+	}
+	return tx, nil
+}
+
+func doSubmitRegister(client grpcRegistry.EntityRegistryClient, signed *entity.SignedEntity) error {
 	req := &grpcRegistry.RegisterRequest{
 		Entity: signed.ToProto(),
 	}
-	if _, err = client.RegisterEntity(context.Background(), req); err != nil {
+	if _, err := client.RegisterEntity(context.Background(), req); err != nil {
 		logger.Error("failed to register entity",
 			"err", err,
 		)
 		return err
 	}
 
-	logger.Info("registered entity",
-		"entity", signer.Public(),
-	)
+	logger.Info("registered entity")
```

Comment:
No need to remove the `entity` argument, just replace the value with `signed.Signature.PublicKey`. (Same below.)